core/utils/assert.hpp

```diff
@@ -0,0 +1,77 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2022 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Valery Mironov
+////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include <cstddef>
+#include <string_view>
+
+#include "shared.hpp"
+
+namespace iresearch {
+
+enum class LogLevel : unsigned char {
```

Comment:
Why not use log levels/log callback from log.hpp/cpp ? We could just add here additional level for assertion if needed (or just use IRL_FATAL)

Reply:
I don't want add new global value

Reply:
Then why  enum? Are we expecting here many asserion log levels?

Reply:
I think maybe in future we remove log with printf format style

Reply:
```cpp
#define LOG_ERROR(cond, message)
 do {
   if (cond) {
     LogMessage(source_location, #cond, message)
   }
 } while(false)


// message will be computed only if condition true
LOG_ERROR(result.isError(), absl::StrCat/Substitute(...)); 
```

Reply:
Still don't get it, why we need separate log levels enum for assertions, with just single value in it. If we will replace printf style logger (I'm totally for it) we will add proper log levele there.